source/solvers/navier_stokes_scratch_data.cc

```diff
@@ -135,6 +138,9 @@ NavierStokesScratchData<dim>::enable_vof(
     std::vector<std::vector<double>>(maximum_number_of_previous_solutions(),
                                      std::vector<double>(this->n_q_points));
   phase_gradient_values = std::vector<Tensor<1, dim>>(this->n_q_points);
+  // For STF calculation
```

Comment:
I wonder, why is there a enable_filtered_vof and an enable_vof now?
Anyway, All of the fields used are the filtered fields no? So why not in the enable_vof just use the filtered field (for phase and for gradient) and if we want to have no filter, we put filter to none?
What do you think @AmishgaAlphonius and @hepap ?
I feel this would make the code simpler. I feel the Navier-Stokes solver should just get the filtered fields and if we disable filtering, we do it everywhere. This way we will prevent confusions like the one we had.

Reply:
To avoid further confusion regarding the "filtered" phase gradient, the old "filtered phase gradient" has been renamed with the term "projected".